### PR TITLE
Add Domains migration command

### PR DESCRIPTION
**Pull request by trasher**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

### CODE REVIEW

File: inc/console/migration/domainsplugintocorecommand.class.php [hunk 1]
```diff
@@ -0,0 +1,745 @@
...
+      if ($items_iterator->count()) {
+         $progress_bar = new ProgressBar($this->output, $items_iterator->count());
+         $progress_bar->start();
+
+         foreach ($items_iterator as $itm) {
+            $progress_bar->advance(1);
+            $core_item = null;
+            $domains_id = $this->getCorrespondingItem('PluginDomainsDomains', $itm['plugin_domains_domains_id'])->fields['id'];
+
+            if (isset($core_items[$domains_id.$itm['itemtype'].$itm['items_id']])) {
```
Comment (cedric-anne) [inline inc/console/migration/domainsplugintocorecommand.class.php]: If relation already exists in core, as you do not update anything except setting `domainrelations_id` to 0, the update part seems useless. I guess just outputing a skipping message and continue to next element will be enough.

File: inc/console/migration/domainsplugintocorecommand.class.php [hunk 2]
```diff
@@ -0,0 +1,745 @@
...
+            $this->outputImportError( __('Unable to create default room.'));
+
+            $room_id = 0;
+         }
+
+         $this->fallback_room_id = $room_id;
+      }
+
+      return $this->fallback_room_id;
+   }*/
```
Comment (cedric-anne) [inline inc/console/migration/domainsplugintocorecommand.class.php]: Can be cleaned.

File: inc/console/migration/domainsplugintocorecommand.class.php [hunk 3]
```diff
@@ -0,0 +1,733 @@
...
+               );
+            }
+            $this->writelnOutputWithProgressBar(
+               $message,
+               $progress_bar,
+               OutputInterface::VERBOSITY_VERY_VERBOSE
+            );
+
+            $mapped_type = $this->getCorrespondingItem('PluginDomainsDomaintype', $dom['plugin_domains_domaintypes_id']);
+            if ($types_id === null) {
```
Comment (cedric-anne) [inline inc/console/migration/domainsplugintocorecommand.class.php]:
```suggestion
            if ($mapped_type === null) {
```

File: inc/console/migration/domainsplugintocorecommand.class.php [hunk 4]
```diff
@@ -0,0 +1,733 @@
...
+               $message = sprintf(
+                  __('Unable to find mapping for type %s.'),
+                  $dom['plugin_domains_domaintypes_id']
+               );
+               $this->outputImportError($message, $progress_bar);
+               return false;
```
Comment (cedric-anne) [inline inc/console/migration/domainsplugintocorecommand.class.php]: I would not consider it as a blocking error, and I would only output it with a normal verbosity

```php
            $this->writelnOutputWithProgressBar(
               $message,
               $progress_bar
            );
```

File: inc/console/migration/domainsplugintocorecommand.class.php [hunk 5]
```diff
@@ -0,0 +1,733 @@
...
+            }
+            $types_id = $mapped_type->fields['id'];
```
Comment (cedric-anne) [inline inc/console/migration/domainsplugintocorecommand.class.php]:
```suggestion
            $types_id = $mapped_type !== null ? $mapped_type->fields['id'] : 0;
```

File: inc/console/migration/domainsplugintocorecommand.class.php [hunk 6]
```diff
@@ -0,0 +1,733 @@
...
+            $domain_input = [
+               'name'                  => $dom['name'],
+               'entities_id'           => $dom['entities_id'],
+               'is_recursive'          => $dom['is_recursive'],
+               'domaintypes_id'        => $types_id ?? 0,
```
Comment (cedric-anne) [inline inc/console/migration/domainsplugintocorecommand.class.php]:
```suggestion
               'domaintypes_id'        => $types_id,
```

File: inc/console/migration/domainsplugintocorecommand.class.php [hunk 7]
```diff
@@ -587,7 +587,7 @@ protected function importDomainItems() {
             if (isset($core_items[$domains_id.$itm['itemtype'].$itm['items_id']])) {
                $core_item = $core_items[$domains_id.$itm['itemtype'].$itm['items_id']];
                $message = sprintf(
-                  __('Updating existing domain item %s...'),
+                  __('Skip existing domain item %s...'),
                   $domains_id . ' ' . $itm['itemtype'] . ' ' . $itm['items_id']
                );
             } else {
```
Comment (cedric-anne) [inline inc/console/migration/domainsplugintocorecommand.class.php]: Adding a `continue` here may help to clean the code on following lines to make it clearer.
```suggestion
               continue;
            } else {
```

Comment (trasher) [reply inc/console/migration/domainsplugintocorecommand.class.php]: With this change, message will not be displayed

Comment (trasher) [reply inc/console/migration/domainsplugintocorecommand.class.php]: See my last commit

**Comment by trasher** [comment]
I've successfully tested with a DB with:
- no types,
- ~2k domains
- a few doamins_items

**Pull request merged**